### PR TITLE
improved mentionsbot v1.0.1

### PR DESCRIPTION
**Pull request by ashfame**
This PR improves the mentions bot plugin, bumping it to v1.0.1

- log exception if one happens when notifying users
- allow groups to define whether triggering message should be quoted or whether mentions should always be in a thread
- move sync config frequency from constant to config

### CODE REVIEW

File: plugins/mentions/mentions.py [hunk 1]
```diff
@@ -119,17 +141,31 @@ async def handle_message(self, evt: MessageEvent) -> None:
...
         body += ", ".join(users_to_notify)
         for user in users_to_notify:
             formatted_body = formatted_body + " <a href='https://matrix.to/#/" + user + "'>" + user + "</a>"
 
-        await self.notify_users(users_to_notify, body, formatted_body, slack_bridge_ignore, evt)
+        await self.notify_users(
+            users_to_notify,
+            body,
+            formatted_body,
+            {
```
Comment (psrpinto) [inline plugins/mentions/mentions.py:158]: Just a detail, but I find it a bit weird that some of the arguments are passed directly, and others are passed as an entry in a dict.

File: plugins/mentions/mentions.py [hunk 2]
```diff
@@ -12,11 +12,21 @@
...
 
 
-class Mentions(Plugin):
-    SYNC_CONFIG_FREQUENCY = 300  # secs
+# returns default bool if all elements in boolList are not same
+def get_collective_behavior(boolList: List[bool], default: bool) -> bool:
+    if len(boolList) == 1:
+        return boolList[0]
+
+    if all(value == boolList[0] for value in boolList):
```
Comment (psrpinto) [inline plugins/mentions/mentions.py:23]: Is it intended to always compare to the first entry in `boolList`? I'm not sure I understand what's going on here.

Comment (ashfame) [reply plugins/mentions/mentions.py:23]: Simply checking if all values are same. So checking all values against the first one.

File: plugins/mentions/mentions.py (hunk 1 above)
Comment (ashfame) [reply plugins/mentions/mentions.py:158]: Not different from defining the last argument as a dict and then passed that variable, right?

**Review by psrpinto — APPROVED** [review]
I think this is fine as is, so I'm approving.

However, I think it might be a bit weird that the plugin behaves differently for different teams. I suspect users will be confused by the fact that sometimes the mention happens in a thread, sometimes not, and that sometimes it quotes the message, sometimes not.

I understand this is the intended behaviour here but, given that the user mentioning the team might not be aware that these things are configurable, they might interpret the difference in behaviour across teams to be a bug.

Personally, I would prefer that the plugin behaves the same always, and I would make it always mention in the thread, since I don't see a use case for having the mention not go in a thread. I would prefer consistency over configurability here. 

That said, as mentioned above, I think this is fine as is.

**Comment by ashfame** [comment]
People of different teams have their preferences of how it should function. Whether the ping should be in a regular message or threaded was voiced in https://github.com/WordPress/matrix/issues/13 already. Hence, configuration on a group level to give everyone the option they would like :)

File: plugins/mentions/mentions.py (hunk 1 above)
Comment (ashfame) [reply plugins/mentions/mentions.py]: done!

**Comment by ashfame** [comment]
Folks in community have strong feelings about threads and would often tell people to not use threads on their first attempt to reply in a thread :)

**Pull request merged**